src/utils/populate.utils.ts

```diff
@@ -0,0 +1,121 @@
+import {mkdirSync} from 'fs';
+import {readFile} from 'fs/promises';
+import {red} from 'kleur';
+import {existsSync} from 'node:fs';
+import {mkdir, readdir, writeFile} from 'node:fs/promises';
+import {dirname, join, relative, resolve} from 'node:path';
+import ora from 'ora';
+import type {GitHubTreeEntry} from '../types/github';
+
+interface FileDescriptor {
+  path: string;
+  url: string;
+}
+
+const REPOSITORY = 'junobuild/create-juno';
+
+const getLocalFiles = async (templatePath: string): Promise<FileDescriptor[]> => {
+  const dirents = await readdir(templatePath, {withFileTypes: true});
+  const files = await Promise.all(
+    dirents.map(async (dirent) => {
+      const res = resolve(templatePath, dirent.name);
+      return dirent.isDirectory() ? await getLocalFiles(res) : res;
+    })
+  );
+  return files
+    .flat()
+    .map((p) =>
+      typeof p === 'string' ? {path: relative(process.cwd(), p), url: `file://${p}`} : p
+    );
+};
+
+const getGitHubFiles = async (templatePath: string): Promise<FileDescriptor[]> => {
+  const res = await fetch(`https://api.github.com/repos/${REPOSITORY}/git/trees/main?recursive=1`);
```

Comment:
We might easily hit some rate limits with the GitHub API given my experience.

Not for this PR, just thinking about it now, should we compress (?, gzip) and deploy the template (CI action) to a "CDN" (assets smart contract/website) and download those from there?

Alternatively, should we include the templates with the CLI?

I guess the latter has the tradeoff of potentially making the CLI quite heavy.

Again, not for this PR, which is already substantial, but for another iteration.

Reply:
Let's use a CDN - i.e. Juno's CDN. I'll setup this.